### DISCUSSION TITLE
Are Inlay hints supported through rust-analyzer?

### DISCUSSION THREAD

**Discussion by AmitPr** [zed-industries/zed#6635] [Feature Requests]
Question as in title. In VSCode I get the nice type hints for function parameters and variables. Is this possible in Zed?

**Comment by JosephTLyons** [comment]
At the time of this comment, inlay hints were not supported, but they have been for awhile now! You can add the following to your `settings.json` file to get inlay hints from `rust-analyzer`.

```json
"lsp": {
  "rust-analyzer": {
    "initialization_options": {
      "inlayHints": {
        "maxLength": null,
        "lifetimeElisionHints": {
          "useParameterNames": true,
          "enable": "skip_trivial"
        },
        "closureReturnTypeHints": {
          "enable": "always"
        }
      }
    }
  }
},
```
